mono/mini/interp/interp.c

```diff
@@ -131,8 +131,11 @@ add_frag (FrameStack *stack, int size)
 {
 	StackFragment *new_frag;
 
-	// FIXME:
-	int frag_size = 4096;
+	// FIXME: The behavior of an allocation not
+	// fitting in current chunk is very poor. Attempt
+	// instead a scheme managing one growable chunk
+	// by using multiple adjacent alloca.
+	int frag_size = 65536;
```

Comment:
nit: The largest portable int power of 2 is 16K
It doesn't have to be a power of 2.
32767 would be ok, or round down for alignment.
31 * 1024 would be good.